### PR TITLE
Feature/eating dirt improvements

### PR DESCRIPTION
**Pull request by JR-Mitchell**
### What?
- Updates `NewSoundEffect` so that the sound effect duration must be passed
- Adds the `Player.Eat_fx` sound effect
- Updates logic in `Player` so that the eat sound effect is played when a block of dirt is eaten
- Updates logic in `Player` so that the player cannot act for a short period after eating a block#50 

### Why?
#20

### CODE REVIEW

File: cart/rabbit.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ const (
 
 func NewRabbit(x, y, mapx, mapy int32, item_sprite *tic80.Sprite) Rabbit {
 	sprite := tic80.SquareSprite(rabbit_main_frame, 4)
-	sfx := tic80.NewSoundEffect(60, 2)
+	sfx := tic80.NewSoundEffect(60, 2, 30)
 	sfx.Duration = 180
```
Comment (fjebaker) [inline cart/rabbit.go]: Just pass `180` here

File: cart/player.go [hunk 2]
```diff
@@ -158,6 +169,21 @@ func (player *Player) Update(t int32, world *World, game *Game) {
 ///////////////////////////////////////////////////////////////////////////////
 // Utils
 
+func (player *Player) startEating(t int32) {
+	player.Eat_fx.Play()
+	player.Eating = true
+	player.EatStartTime = t
+}
+
+func (player *Player) isEating(t int32) bool {
```
Comment (fjebaker) [inline cart/player.go]: Maybe clearer to call that `IsStillEating`?

**Review by fjebaker — CHANGES_REQUESTED** [review]

File: cart/rabbit.go (hunk 1 above)
Comment (JR-Mitchell) [reply cart/rabbit.go]: Done in 940a7e0

File: cart/player.go (hunk 2 above)
Comment (JR-Mitchell) [reply cart/player.go]: Updated in 030ec4b to call it `updateEatingState` and not have it return a value, instead checking `player.Eating` straight after the call.

**Review by fjebaker — APPROVED** [review]

**Pull request merged**